contracts/core/proofAPI/CrossL2Prover.sol

```diff
@@ -31,29 +31,31 @@ contract CrossL2Prover is AppStateVerifier, ICrossL2Prover {
     LightClientType public constant LIGHT_CLIENT_TYPE = LightClientType.SequencerLightClient; // Stored as a constant
         // for cheap on-chain use
 
-    string public peptideClientId;
+    bytes32 public clientType;
     ISignatureVerifier public immutable verifier;
 
     // Store peptide apphashes for a given height
     mapping(uint256 => uint256) public peptideAppHashes;
 
     error CannotUpdateClientWithDifferentAppHash();
 
-    constructor(ISignatureVerifier verifier_, string memory peptideClientId_) {
+    constructor(ISignatureVerifier verifier_, bytes32 clientType_) {
         verifier = verifier_;
-        peptideClientId = peptideClientId_;
+        clientType = clientType_;
     }
 
     /**
      * @inheritdoc ICrossL2Prover
      */
-    function validateReceipt(bytes calldata receiptIndex, bytes calldata receiptRLPEncodedBytes, bytes calldata proof)
-        public
-        view
-        returns (bool valid)
-    {
-        (Ics23Proof memory peptideAppProof, bytes[] memory receiptMMPTProof, bytes32 receiptRoot, uint64 eventHeight) =
-            abi.decode(proof, (Ics23Proof, bytes[], bytes32, uint64));
+    function validateReceipt(bytes calldata proof) public view returns (bytes32 srcChainID, bytes memory receiptRLP) {
+        (
+            Ics23Proof memory peptideAppProof,
+            bytes[] memory receiptMMPTProof,
+            bytes32 receiptRoot,
+            uint64 eventHeight,
+            bytes32 srcChainId,
```

Comment:
this needs to match what peptide is doing: https://github.com/polymerdao/relayer/pull/17/files#diff-272397fd750808e0c4740b9200992b4bc15a0cacfd75cb4360bef55ef743b43eR203
ie it should be a string encoded number

Reply:
done!